[ROX-27310]: New page: Understanding the differences in Node CVEs between the Stackrox Scanner and Scanner V4 (#88222)
openshift/openshift-docs: modules/understanding-node-cves-scanner-v4.adoc

// Module included in the following assemblies:
//
// * operating/manage-vulnerabilities/common-vuln-management-tasks.adoc

:_mod-docs-content-type: CONCEPT
// DO NOT RENAME THE ID - there are UI links using it in ACS
[id="understanding-node-cves-scanner-v4_{context}"]
= Understanding differences in scanning results between the Stackrox Scanner and Scanner V4

> == Summary

Reviewer: I'm sorry, I missed this before - we aren't supposed to use headings like "Overview" or "Summary". I think we can actually just remove this because what follows is the introductory text - see the [mod docs guidance](https://redhat-documentation.github.io/modular-docs/#creating-concept-modules) (but to summarize - we don't need a heading here).

> Scanning {op-system} node hosts with Scanner V4 reports significantly more CVEs (~390) compared to StackRox Scanner (~50) for the same operating system version. A manual review of selected vulnerabilities revealed that:

Author: I spotted one more issue here - the first sentence uses `more` but has no `than`.
So I changed the first sentence to:
``` 
Scanning {op-system} node hosts with Scanner V4 reports significantly more CVEs for the same operating system version than the Stackrox Scanner.
```

* The VEX data used in Scanner V4 is more accurate.
* The VEX data includes granular statuses, such as "no fix planned" and "fix deferred".
* Some vulnerabilities reported by StackRox Scanner were false positives.

As a result, Scanner V4 provides a more accurate and realistic vulnerability assessment.

Users may find discrepancies in reported vulnerabilities surprising—especially if some secured clusters still use older {product-title-short} versions with StackRox Scanner while others have transitioned to Scanner V4. This article explains these differences and provides guidance on how to manually verify reported vulnerabilities if needed.

== Discrepancies in reported vulnerabilities

We analyzed the differences in reported CVEs for three recent, arbitrarily selected {op-system} versions. This article presents findings for {op-system} version `417.94.202501071621-0`.
For this version:

* StackRox Scanner reports 49 CVEs.

> * Scanner V4 reports 389 CVEs.

Reviewer: ```suggestion
* Scanner V4 reported 389 CVEs.
```

The breakdown is as follows:

* 1 CVE is reported only by the StackRox Scanner.
* 48 CVEs are reported by both scanners.
* 341 CVEs are reported only by Scanner V4.

=== CVEs reported only by the StackRox Scanner

The single CVE reported exclusively by StackRox Scanner was a false positive. CVE-2022-4122 was flagged for the `podman` package in version `5:5.2.2-1.rhaos4.17.el9.x86_64`. However, a manual review of VEX data from link:https://access.redhat.com/errata/RHSA-2024:9102[RHSA-2024:9102] indicated that this vulnerability was fixed in version `5:5.2.2-1.el9`. Therefore, the package version scanned was the first to contain the fix and was no longer affected.

=== CVEs reported only by Scanner V4

We randomly selected 10 CVEs from the 341 unique to Scanner V4 and conducted a detailed analysis using VEX data. The vulnerabilities fell into two categories:

> . Affected packages with a fine-grained status indicating that no fix is planned.

Reviewer: ```suggestion
* Affected packages with a fine-grained status indicating that no fix is planned
```

> . Affected packages with no additional status details regarding a fix.

Reviewer: ```suggestion
* Affected packages with no additional status details regarding a fix
```
These don't need to be numbered because the [sequence isn't significant](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#unordered-lists) and they are not complete sentences so do not need punctuation - see [punctuation](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#punctuation-in-lists)

> For example:

Reviewer: ```suggestion
For example, the following results were analyzed:
```

> * The `git-core` package (version `2.43.5-1.el9_4`) was flagged for CVE-2024-50349 (link:https://access.redhat.com/security/cve/cve-2024-50349[VEX data]), marked as "Affected" with a fine-grained status of "Fix deferred." This means a fix is not guaranteed due to higher-priority development work. The package is affected by three CVEs in total.

Reviewer: ```suggestion
* The `git-core` package (version `2.43.5-1.el9_4`) was flagged for CVE-2024-50349 (link:https://access.redhat.com/security/cve/cve-2024-50349[VEX data]) and marked as "Affected" with a fine-grained status of "Fix deferred." This means a fix is not guaranteed due to higher-priority development work. The package is affected by three CVEs in total.
```

* The `vim-minimal` package (version `2:8.2.2637-20.el9_1`) was flagged for 109 CVEs, 108 of which have low CVSS scores. Most are marked as "Affected" with a fine-grained status of "Will not fix."

> * The `krb5-libs` package (version `1.21.1-2.el9_4.1`) was flagged for CVE-2025-24528 (link:https://access.redhat.com/security/cve/cve-2025-24528[VEX data]), but no fine-grained status was available. Given that this CVE was recently discovered (2025), its status may be updated soon.

Reviewer: ```suggestion
* The `krb5-libs` package (version `1.21.1-2.el9_4.1`) was flagged for CVE-2025-24528 (link:https://access.redhat.com/security/cve/cve-2025-24528[VEX data]), but no fine-grained status was available. Given that this CVE was recently discovered at the time of this analysis, its status might be updated soon.
```

=== CVEs reported by both scanners

> This category aligns with expectations. We manually verified three randomly selected packages, finding that OVAL2 data (used in the StackRox Scanner) and VEX data (used in Scanner V4) provided consistent explanations for the detected CVEs. In some cases, CVSS scores differed slightly, which is expected due to variations in VEX publisher data.

Reviewer: ```suggestion
We manually verified three randomly selected packages, finding that the OVAL v2 data used in the StackRox Scanner and the VEX data used in Scanner V4) provided consistent explanations for the detected CVEs. In some cases, CVSS scores differed slightly, which is expected due to variations in VEX publisher data.
```
Suggested removing "aligns with expectations" but if you feel strongly about it, it can stay. :)

Author: no strong feelings :) I take all the guidance, as the consistency with the rest of the docs is much more important in this case, and I have no habits regarding IBM style (I think, I still cannot access the page that describes it).

> == Recommendation

Reviewer: ```suggestion
== Verifying the status of vulnerabilities
```

> We recommend manually verifying the fine-grained statuses of vulnerabilities in node host components that are critical to your environment using publicly available VEX data. VEX data is accessible in both human-readable and machine-readable formats. More information on interpreting VEX data is available in this link:https://www.redhat.com/en/blog/recent-improvements-red-hat-enterprise-linux-coreos-security-data[Red Hat Blog].

Reviewer: ```suggestion
As a best practice, verify the fine-grained statuses of vulnerabilities in node host components that are critical to your environment using publicly available VEX data. VEX data is accessible in both human-readable and machine-readable formats. For more information about interpreting VEX data, visit link:https://www.redhat.com/en/blog/recent-improvements-red-hat-enterprise-linux-coreos-security-data[Recent improvements in Red Hat Enterprise Linux CoreOS security data].
```
We are supposed to [avoid using "recommend"](https://redhat-documentation.github.io/supplementary-style-guide/#recommend) - tried to rewrite slightly.
